overlord/snapshotstate/backend/backend.go

```diff
@@ -229,27 +229,68 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 var isTesting = osutil.GetenvBool("SNAPPY_TESTING")
 
 func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer, username string, entry, dir string) error {
-	hasher := crypto.SHA3_384.New()
-	if exists, isDir, err := osutil.DirExists(dir); !exists || !isDir || err != nil {
-		if exists && !isDir {
+	parent, revdir := filepath.Split(dir)
+	if exists, isDir, err := osutil.DirExists(parent); !(err == nil && exists && isDir) {
+		if err != nil {
+			return err
+		}
+		if exists {
+			logger.Noticef("Not saving directories under %q in snapshot #%d of %q as it is not a directory.", parent, snapshot.SetID, snapshot.Snap)
+		} else {
+			logger.Debugf("Not saving directories under %q in snapshot #%d of %q as it is does not exist.", parent, snapshot.SetID, snapshot.Snap)
+		}
+		return nil
+	}
+	tarArgs := []string{
+		"--create",
+		"--sparse", "--gzip",
+		"--directory", parent,
+	}
+
+	noRev, noCommon := true, true
+
+	if exists, isDir, err := osutil.DirExists(dir); err == nil && exists && isDir {
```

Comment:
Similiar to earlier comment, would it make sense to get rid of extra `if` level here and simply check err first?